### PR TITLE
fix release workflow

### PR DESCRIPTION
**Pull request by r-ashish**
Follow up fix for https://github.com/gomods/athens/pull/1823

### CODE REVIEW

**Comment by r-ashish** [comment]
Tested and confirmed that both canary and commit tags are being pushed: 

- https://hub.docker.com/r/gomods/athens/tags
- https://hub.docker.com/r/gomods/athens-dev/tags

**Review by manugupt1 — APPROVED** [review]
Thanks!

File: .github/workflows/release.latest.yml [hunk 1]
```diff
@@ -21,3 +21,4 @@ jobs:
           file: cmd/proxy/Dockerfile
           build-args: VERSION=${{github.sha}}
           tags: gomods/athens:latest
+          push: true
```
Comment (nrwiersma) [inline .github/workflows/release.latest.yml:24]: We are not pushing a tagged image here, only latest, which no one uses in production.

Comment (r-ashish) [reply .github/workflows/release.latest.yml:24]: Right, will update later. Idk how it was working in drone though 🤔. (maybe I missed a step while migrating)

Comment (nrwiersma) [reply .github/workflows/release.latest.yml:24]: https://github.com/gomods/athens/commit/3f26845cffd789688277c8a79d1cc11159a2c689#diff-b54b39f1afced2465e1f3641db9d5bbf4f3a7fcf890996dfedd3c197bcb7f8c7L123

Comment (r-ashish) [reply .github/workflows/release.latest.yml:24]: Updated. Thanks for pointing out!

**Review by nrwiersma — APPROVED** [review]
LGTM 🎉

**Pull request merged**